File: game/src/main/java/io/github/fernthedev/secondgame/main/HUD.java
//
// Source code recreated from a .class file by IntelliJ IDEA
// (powered by Fernflower decompiler)
//

package io.github.fernthedev.secondgame.main;

import java.awt.*;
import java.util.concurrent.TimeUnit;

public class HUD {
    private int greenvalue = 255;
    private int score = 0;
    private int level = 1;

    public HUD() {
    }

    public void tick() {

        if (Game.getMainPlayer() != null) {
            Game.getMainPlayer().setHealth((int) Game.clamp((float) Game.getMainPlayer().getHealth(), 0.0F, 100.0F));
            //  System.out.println(GAME.mainPlayer + " " + Game.mainPlayer.getHealth());
            Game.getMainPlayer().setHealth(Game.getMainPlayer().getHealth());


            this.greenvalue = (int) Game.clamp((float) this.greenvalue, 0.0F, 255.0F);
            this.greenvalue = Game.getMainPlayer().getHealth() * 2;
            this.score++;
        }


//        if(Game.gameState == Game.STATE.GAME) {
//            Game.mainPlayer.setHealth( (int)Game.clamp((float) Game.mainPlayer.getHealth(), 0.0F, 100.0F) );
//            //  System.out.println(GAME.mainPlayer + " " + Game.mainPlayer.getHealth());
//            if (Game.mainPlayer != null)
//                Game.mainPlayer.setHealth(Game.mainPlayer.getHealth());
//
//            this.greenvalue = (int) Game.clamp((float) this.greenvalue, 0.0F, 255.0F);
//            this.greenvalue = Game.mainPlayer.getHealth() * 2;
//            this.score++;
//        }
//
//        if(Game.gameState == Game.STATE.IN_SERVER || Game.gameState == Game.STATE.HOSTING) {
//            if(Game.mainPlayer != null) {
//                Game.mainPlayer.setHealth(Game.mainPlayer.getHealth());
//
//                Game.mainPlayer.setHealth( (int) Game.clamp((float) Game.mainPlayer.getHealth(), 0.0F, 100.0F));
//
//
//                this.greenvalue = (int) Game.clamp((float) this.greenvalue, 0.0F, 255.0F);
//                this.greenvalue = Game.mainPlayer.getHealth() * 2;
//            }
//        }
    }

    public void render(Graphics g) {
        if (Game.getMainPlayer() != null) {
            g.setColor(Color.GRAY);
            g.fillRect(15, 15, 200, 32);
            g.setColor(new Color(75, this.greenvalue, 0));
            g.fillRect(15, 15, Game.getMainPlayer().getHealth() * 2, 32);
            if (Game.getMainPlayer().getHealth() == 0) {
                g.setColor(Color.RED);
                g.drawRect(15, 15, 200, 32);
            } else {
                g.setColor(Color.WHITE);
                g.drawRect(15, 15, 200, 32);
            }

            g.setColor(Color.WHITE);
            g.drawString("Score: " + this.score, 15, 64);
            g.drawString("Level: " + this.level, 15, 80);
            g.drawString("Coins: " + Game.getMainPlayer().getCoin(), 15, 96);
            if (Game.getClient() != null) {
                try {
                    g.drawString("Ping: " + Game.getClient().getPingTime(TimeUnit.MILLISECONDS) + "ms", 15, 112);
                } catch (RuntimeException e) {
                    e.printStackTrace();
                }
            }
        }
    }

    public int getScore() {
        return this.score;
    }

    public void setScore(int score) {
        this.score = score;
    }

    public int getLevel() {
        return this.level;
    }

    public void setLevel(int level) {
        this.level = level;
    }
}

File: game/src/main/java/io/github/fernthedev/secondgame/main/HUD.kt
//
// Source code recreated from a .class file by IntelliJ IDEA
// (powered by Fernflower decompiler)
//
package io.github.fernthedev.secondgame.main


import io.github.fernthedev.secondgame.main.ui.api.ScreenFont
import java.awt.Color
import java.awt.Font
import java.awt.Graphics2D
import java.util.concurrent.TimeUnit

class HUD {
    private var greenvalue = 255
    var score = 0
    var level = 1

    val textFont: ScreenFont = ScreenFont(Font("arial", Font.PLAIN, 12), Color.WHITE)
    fun tick() {
        if (Game.mainPlayer == null) return

        greenvalue = Game.clamp(greenvalue.toFloat(), 0.0f, 255.0f).toInt()
        greenvalue = Game.mainPlayer!!.health * 2
        score++
    }

    fun render(g: Graphics2D) {
        g.font = textFont.font

        val mainPlayer = Game.mainPlayer

        if (mainPlayer != null) {
            g.color = Color.GRAY
            g.fillRect(15, 15, 200, 32)

            g.color = Color(75, greenvalue, 0)
            g.fillRect(15, 15, mainPlayer.health * 2, 32)

            if (mainPlayer.health == 0) {
                g.color = Color.RED
                g.drawRect(15, 15, 200, 32)
            } else {
                g.color = Color.WHITE
                g.drawRect(15, 15, 200, 32)
            }

            g.color = Color.WHITE
            g.drawString("Coins: " + mainPlayer.coin, 15, 96)
        }
        
        g.color = Color.WHITE
        g.drawString("Score: $score", 15, 64)
        g.drawString("Level: $level", 15, 80)

        if (Game.client != null) {
            try {
                g.drawString("Ping: " + Game.client!!.getPingTime(TimeUnit.MILLISECONDS) + "ms", 15, 112)
            } catch (e: RuntimeException) {
                e.printStackTrace()
            }
        }
    }
}